Improving debugging workshop (#496)
NuevoFoundation/workshops: content/english/debugging/04-identify-the-problem3.md

```
{{% /expand %}}
{{% expand "***Answer***" %}}

> The `p1` variable is not being increased when a number lesser than the pivot is found. Remember that our version will assume that each time a lesser element is found, the pointer for the greater element must be updated.

Reviewer: ```suggestion
The `p1` variable is not being increased when a number smaller than the pivot is found. Remember that our version will assume that each time a lesser element is found, the pointer for the greater element must be updated.

```

Note that all debuggers should have the same basic concepts: they allow you to step through your code as it runs, in real-time. You should almost always use debuggers over print statements. You won't regret it!

### GDB Commands
There are plenty of commands GDB offers. This table summarizes some useful ones, but we also recommend using this [GDB Reference Sheet](https://cs.brown.edu/courses/cs033/docs/guides/gdb.pdf) to see more commands you can use.

| Command | What It Does |
|---|---|
| `gdb <name of program>` | Runs/loads the program with GDB |
| `run` or `r` | Runs the loaded executable program |
| `quit` or `q` | Exits GDB |

> | `break <filename:line>` | Create a breakpoint |

Reviewer: ```suggestion
| `break <filename:line>` | Create a breakpoint at the specified file and line number |

```

| `print <variable name>` or `print <expression>` | Print the value of an expression which can contain variable names and constants |
| `next` or `n` | Advance to the next line of code without jumping into a function call |
| `step` | Move to the first line of code in a function call | 
| `continue` or `c` | Continue running your code until the next breakpoint | 

> | `delete <breakpoint #>` or `d <breakpoint #>` | Delete a breakpoint at the # |

Reviewer: ```suggestion
| `delete <breakpoint #>` or `d <breakpoint #>` | Delete a breakpoint at line number # |

```
